agent/dns.go

```diff
@@ -1509,7 +1515,10 @@ func (d *DNSServer) handleServiceQuery(cfg *dnsRequestConfig, lookup serviceLook
 	// Add various responses depending on the request
 	qType := req.Question[0].Qtype
 	if qType == dns.TypeSRV {
-		d.addServiceSRVRecordsToMessage(cfg, lookup, out.Nodes, req, resp, ttl, lookup.MaxRecursionLevel)
+		err := d.addServiceSRVRecordsToMessage(cfg, lookup, out.Nodes, req, resp, ttl, lookup.MaxRecursionLevel)
```

Comment:
This is change of logic, we may want to return DNS response even in case of error. Avoid changing logic in this PR.

Reply:
We do return a DNS response in case of error. The response varies based on the error returned and is handled for us.